diff --git a/main/env.py b/main/env.py
index d18a8a9..5aab599 100644
--- a/main/env.py
+++ b/main/env.py
@@ -202,7 +202,7 @@ class DiffDiscreteEventSystem(gym.Env):
         #if self.f_preemptive:
         ###TODO Changed
         # action = torch.minimum(action, queues)
-        action = torch.minimum(action, queues.unsqueeze(1).repeat(1,self.s,1))
+        action = torch.minimum(action, queues.unsqueeze(1).expand(-1,self.s,-1))
 
         # work is action times mu
 
@@ -333,7 +333,7 @@ class DiffDiscreteEventSystem(gym.Env):
 
         for q in range(self.q):
             if num_allocated[q] > 0:
-                service_times[q][:num_allocated[q]] = list(torch.unbind(torch.stack(service_times[q][:num_allocated[q]]) - event_time * torch.stack(allocated_work[q]).unsqueeze(1).unsqueeze(-1).repeat((1,1,self.network.shape[-1]))))
+                service_times[q][:num_allocated[q]] = list(torch.unbind(torch.stack(service_times[q][:num_allocated[q]]) - event_time * torch.stack(allocated_work[q]).unsqueeze(1).unsqueeze(-1).expand(-1,-1,self.network.shape[-1])))
                 # service_times[q][:nuqueuem_allocated[q]] = list(torch.unbind(torch.stack(service_times[q][:num_allocated[q]]) - event_time.detach() * torch.stack(allocated_work[q]).detach().unsqueeze(1).unsqueeze(-1).repeat((1,1,self.network.shape[-1]))))
         # update arrival times
         # if torch.min(arrival_times) == 0:
