diff --git a/main/env.py b/main/env.py
index 5aab599..fdf9600 100644
--- a/main/env.py
+++ b/main/env.py
@@ -401,7 +401,7 @@ class DiffDiscreteEventSystem(gym.Env):
 
         info = {"obs": obs, "state": next_state, "cost": cost, "event_time": event_time, "queues": queues}
 
-        return queues, reward, done, truncated, info
+        return queues.cpu().detach().numpy(), reward, done, truncated, info
     
     def get_observation(self):
         return self.queues
