diff --git a/RL/utils/rl_env.py b/RL/utils/rl_env.py
index ce83bbc..2db4ad1 100644
--- a/RL/utils/rl_env.py
+++ b/RL/utils/rl_env.py
@@ -107,10 +107,9 @@ class RL_Wrapper_P_DiffDiscreteEventSystem(DiffDiscreteEventSystem):
         return queues.cpu().numpy(), {}
     
     def reset_env_seed(self):
-        seed = self.seed
-        # print(f"seed: {seed}")  
-        if seed is not None:
-            self.state = np.random.RandomState(seed)
+        if self.seed is not None:
+            self.state = torch.Generator(device=self.device)
+            self.state.manual_seed(self.seed)
 
 
 def load_rl_p_env(env_config, temp, batch, seed, policy_name, device):
diff --git a/main/env.py b/main/env.py
index 679699b..bd913d2 100644
--- a/main/env.py
+++ b/main/env.py
@@ -131,7 +131,9 @@ class DiffDiscreteEventSystem(gym.Env):
         
 
         self.device = device
-        self.state = np.random.default_rng(seed)
+        self.state = torch.Generator(device=self.device)
+        if seed is not None:
+            self.state.manual_seed(seed)
         self.network = network.repeat(batch,1,1).to(self.device)
         self.mu = mu.repeat(batch,1,1).to(self.device)
         self.q = self.network.size()[-1]
@@ -196,7 +198,8 @@ class DiffDiscreteEventSystem(gym.Env):
             queues = init_queues.float().to(self.device)
         
         if seed is not None:
-            self.state = np.random.RandomState(seed)
+            self.state = torch.Generator(device=self.device)
+            self.state.manual_seed(seed)
 
         # Tensor service times: st_data[Q, max_jobs], st_counts[Q]
         self.st_counts = (queues[0].int() if self.batch == 1 else queues[:, 0].int()).to(self.device)
