diff --git a/RL/utils/rl_env.py b/RL/utils/rl_env.py
index 46e2777..35ea1c5 100644
--- a/RL/utils/rl_env.py
+++ b/RL/utils/rl_env.py
@@ -7,7 +7,7 @@ print(project_root)
 import numpy as np
 import torch
 from gym import spaces
-from main.env import DiffDiscreteEventSystem
+from main.env import BatchedEnv, DiffDiscreteEventSystem
 from typing import NamedTuple
 
 
@@ -230,15 +230,16 @@ def load_rl_p_env(env_config, temp, batch, seed, policy_name, device):
         
     #     return interarrivals
 
-    dq = RL_Wrapper_P_DiffDiscreteEventSystem(network, mu, h, 
-                                       draw_service= draw_service, draw_inter_arrivals = draw_inter_arrivals, init_time = 0, 
-                                    queue_event_options= queue_event_options,
-                                    batch = batch, 
-                                    temp = temp, seed = seed,
-                                    time_f = False,
-                                    reward_scale = 1.0,
-                                    policy_name= policy_name,
-                                    action_map = None,
-                                    device = torch.device(device))
-
-    return dq
\ No newline at end of file
+    env_cls = BatchedEnv if batch > 1 else RL_Wrapper_P_DiffDiscreteEventSystem
+    dq = env_cls(network, mu, h,
+                 draw_service=draw_service, draw_inter_arrivals=draw_inter_arrivals, init_time=0,
+                 queue_event_options=queue_event_options,
+                 batch=batch,
+                 temp=temp, seed=seed,
+                 time_f=False,
+                 reward_scale=1.0,
+                 policy_name=policy_name,
+                 action_map=None,
+                 device=torch.device(device))
+
+    return dq
